hydra_base/lib/template.py

```diff
@@ -174,22 +206,25 @@ def parse_xml_typeattr(type_i, attribute):
     if attribute.find('data_type') is not None:
         typeattr_i.data_type = attribute.find('data_type').text
 
+    # Analyzing the "default" node
     if attribute.find('default') is not None:
         default = attribute.find('default')
         unit = default.find('unit').text
+        unit_id = units.get_unit_by_abbreviation(unit).id
 
-        if unit is None and typeattr_i.unit is not None:
-            unit = typeattr_i.unit
+        if unit_id is None and typeattr_i.unit_id is not None:
+            unit_id = typeattr_i.unit_id
 
-        dimension = None
-        if unit is not None:
-            _check_dimension(typeattr_i, unit)
-            dimension = units.get_unit_dimension(unit)
+        dimension_id = None
+        if unit_id is not None:
+            _check_dimension(typeattr_i, unit_id)
+            #dimension = units.get_unit_dimension(unit)
+            dimension_id = units.get_dimension_by_unit_id(unit_id).id
 
-        if unit is not None and typeattr_i.unit is not None:
-            if unit != typeattr_i.unit:
+        if unit_id is not None and _exists_key_in_object("unit_id",JSONObject(typeattr_i)) and typeattr_i.unit_id is not None:
```

Comment:
see my comment above. Remove this function call.

Reply:
I cannot remove it. Read my comment please.